src/metadata/metadata_hash.c

```diff
@@ -62,7 +62,7 @@ struct ocf_metadata_hash_ctrl {
 	ocf_cache_line_t cachelines;
 	ocf_cache_line_t start_page;
 	ocf_cache_line_t count_pages;
-	uint32_t device_lines;
+	uint64_t device_lines;
```

Comment:
After changing type of `device_lines` to `uint64_t`, assignment in line 242 does value narrowing conversion to `int64_t`, which makes it possible for `cache_lines` to take negative values if cache device is huge enough.

Reply:
Nice catch. I've changed `cache_lines` to `uint64_t` since it shouldn't ever go below zero.

Reply:
So now it' the same about `diff_lines` and `lowest_diff`. Moveover in case of `diff_lines` we have possible multiplication overflow in line 298.

Reply:
True that. I've reverted to previous version where we just check if we won't overflow on `device_lines`. This means there is a narrow window of device sizes which won't be supported even though they could. I think this close to the release I'm not brave enough to tune the `(...)calculate_metadata_size` function. Maybe next time? We should consider expanding ocf_cacheline_t anyways.